awx/main/models/jobs.py

```diff
@@ -767,19 +774,14 @@ def is_container_group_task(self):
 
     @property
     def preferred_instance_groups(self):
-        if self.organization is not None:
-            organization_groups = [x for x in self.organization.instance_groups.all()]
-        else:
-            organization_groups = []
-        if self.inventory is not None:
-            inventory_groups = [x for x in self.inventory.instance_groups.all()]
-        else:
-            inventory_groups = []
-        if self.job_template is not None:
-            template_groups = [x for x in self.job_template.instance_groups.all()]
-        else:
-            template_groups = []
-        selected_groups = template_groups + inventory_groups + organization_groups
+        selected_groups = []
+        for obj_type in ['job_template', 'inventory', 'organization']:
+            if getattr(self, obj_type) is not None:
+                for instance_group in getattr(self, obj_type).instance_groups.all():
+                    selected_groups.append(instance_group)
+                if getattr(getattr(self, obj_type), 'prevent_instance_group_fallback', False):
+                    logger.error("Breaking in preferred instance group at {}".format(obj_type))
```

Comment:
Since this is intended, I don't think a log would be appropriate. Even if you do a different lower-level log, there's contextual info we'd need, like the job id.

Reply:
Good catch, this was intended as debug but I bumped it up for testing. I'll move this back to debug and add a job_id and add more info like job.id for clarity.

Reply:
Actually, I can't add the job ID here because we are still in the creation of the job. We could say something like:
```
For job template <name> (<id>) breaking in preferred instance group at <obj_type> when launching.
``` 
Thoughts? Should we just get rid of the log message?

Reply:
Yeah, that's an option. You could even do the JT id as a less risk-prone option. But I'd also say that getting rid of the log would be a good idea.

Reply:
Debug removed per our conversation.